provisioning.adoc

```diff
@@ -0,0 +1,303 @@
+instance is responsible for invoking and co-ordinating all the provisioning
+flows. Any node-local executions can be pushed out for the node agent residing
+on those specific nodes to execute, by the Provisioning Controller. An example
+of such a flow is the installation of the sds bridge on specific nodes of the
+cluster using the Integration Module. The Provisioning Controller would also be
+responsible for installing and interfacing with the Provisioner, locally, using
+the Provisioner Interface Module.
+
+
+== Provisioning Scenarios
+
+Tendrl currently needs to invoke a provisioning system to be able to:
+* deploy various storage system components
+* detect the version of an installed and running storage system
+* deploy storage system specific tendrl-sds bridge
+
+This list could be extended in the future to include:
+* additional tendrl stacks
+* storage system specific collectd plugins for the performance monitoring
+stack
+
+Additional requirements could also be added in the future.
+
+There are three provisioning scenarios, based on the layers of provisioning,
+that tendrl needs to address:
+
+Native:: Tendrl itself ships with and is able to natively execute ansible
+playbooks.
+Internal:: Tendrl invokes an ssh based deployment system by setting it up
+first.
+External:: Tendrl invokes a completely external system that it does not need to
+do any kind of setup for and is self-sufficient. This scenario includes systems
+such as puppet or satellite.
+
+NOTE: The terms Internal and External are used here in the context of a
+deployment. This has nothing to do with them being integrated with tendrl as a
+codebase or packaged with tendrl etc. For tendrl as a codebase, the Internal
+systems are still external integrations. They're simply deployed in the same
+deployment environment as the storage system and tendrl itself.
+
+Following are the scopes of these scenarios:
+Native:: Node-local only.
+Internal:: Cluster specific.
+External:: Multi-cluster.
+
+
+=== Native Execution Scenario
+
+In this scenario, tendrl’s node agent would be able to:
+* Execute shell commands directly.
+* Execute ansible playbooks, specifically the Provisioner Provisioning and
+Integration Modules.
+
+Node agent is responsible for handling any interactions with the operating
+system. Node agent is a prerequisite for a working tendrl core stack and is
+guaranteed to be present on every node of the deployment. It is, hence, the
+natural fit for the native execution scenario.
+
+NOTE: Every node agent is able to execute the Native Execution Scenario.
+
+
+=== Internal Execution Scenario
+
+In this scenario, tendrl depends upon Provisioners to setup the storage system.
+Internal execution scenario has the following advantages:
+* The framework capabilities required to be developed for tendrl should also be
+able to serve the external execution scenario.
+* The provisioning systems for both ceph and gluster are already mature and
```

Comment:
If you are referring ceph-ansible for ceph its fine. But ceph-installer, is available only on RHEL till now. Just an alarm :)